src/tir/transforms/storage_rewrite.cc

```diff
@@ -661,9 +661,11 @@ class StoragePlanRewriter : public StmtExprMutator {
                                   e->allocs[0]->condition, Evaluate(0));
           if (IsSpecialTaggedMemory(e->scope)) {
             MemoryInfo info = GetMemoryInfo(e->scope.to_string());
-            uint64_t total_elem = e->const_nbits / e->elem_type.bits();
-            ICHECK_LE(total_elem * e->elem_type.bits(), info->max_num_bits)
-                << "Allocation exceed bound of memory tag " << e->scope.to_string();
+            if (info.defined()) {
+              uint64_t total_elem = e->const_nbits / e->elem_type.bits();
+              ICHECK_LE(total_elem * e->elem_type.bits(), info->max_num_bits)
+                  << "Allocation exceed bound of memory tag " << e->scope.to_string();
+            }
```

Comment:
shall we throw a warning / error if memory info is not defined?

Reply:
As structured, MemoryInfo is not well defined for Nd-memory like global.texture. If MemoryInfo is a concept we want to keep around (it's being used in two places and there exists only one definition of a MemoryInfo in the repo), then we should consider an additional PR to update its semantics. In it's current state, a warning would be applicable, I did not do that because MemoryInfo is used in other places in storage rewrite without issuing a warning when undefined and guarded with `if (info.defined()) {`.

I can update GetMemoryInfo to issue a warning when it returns a default constructed MemoryInfo().

Reply:
That makes sense, thanks for the explanation. I don’t have a preference.

Reply:
One of these places is `python/tvm/contrib/hexagon/transform.py` and it can definitely be removed from there.